server/services/nl.py

```diff
@@ -58,31 +60,32 @@ def _download_embeddings(self):
       _, filename = os.path.split(blob.name)
       blob.download_to_filename(os.path.join(TEMP_DIR, filename))  # Download
 
-  def detect_svs(self, query):
+  def detect_svs(self, query, embeddings_build):
     query_embeddings = self.model.encode([query])
+    if embeddings_build not in self.dataset_embeddings_maps:
+      return {'SV': [], 'CosineScore': []}
     hits = semantic_search(query_embeddings,
-                           self.dataset_embeddings_maps[BUILD],
-                           top_k=15)
+                           self.dataset_embeddings_maps[embeddings_build],
+                           top_k=10)
 
     # Note: multiple results may map to the same DCID. As well, the same string may
     # map to multiple DCIDs with the same score.
     sv2score = {}
-    score2svs = {}
     for e in hits[0]:
-      for d in self.dcid_maps[BUILD][e['corpus_id']].split(','):
+      for d in self.dcid_maps[embeddings_build][e['corpus_id']].split(','):
         s = e['score']
         # Prefer the top score.
         if d not in sv2score:
           sv2score[d] = s
-          if s not in score2svs:
```

Comment:
This was resulting in a bug and the code change below fixes it. The bug was that the svs where the score matched exactly were returned together with a " : " separator. Instead, we should be returning all SVs separately.